mypy/plugins/dataclasses.py

```diff
@@ -197,6 +202,34 @@ def transform(self) -> None:
             'frozen': decorator_arguments['frozen'],
         }
 
+    def add_slots(self,
+                  info: TypeInfo,
+                  attributes: List[DataclassAttribute],
+                  correct_version: bool) -> None:
+        if not correct_version:
+            # This means that version is lower than `3.10`,
+            # it is just a non-existent argument for `dataclass` function.
+            self._ctx.api.fail(
+                'Unexpected keyword argument "slots" for "dataclass"',
```

Comment:
```suggestion
                'Keyword argument "slots" for "dataclass" is only valid in Python 3.10 and higher',
```
This will be more helpful for users who run into this because they didn't configure `--python-version` correctly.